### PR TITLE
✨ Finalize network v1a2 naming convention

### PR DESCRIPTION
**Pull request by sreyasn**
**What does this PR do, and why is it needed?**
This change checks for v1a1 network named interfaces before creating or updating netIfs with the new naming convention.

**Which issue(s) is/are addressed by this PR?** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

Fixes # N/A

**Are there any special notes for your reviewer**:

N/A

**Please add a release note if necessary**:

```
N/A
```

### CODE REVIEW

**Review by dilyar85 — APPROVED** [review]
LGTM.

File: pkg/vmprovider/providers/vsphere2/network/network.go [hunk 1]
```diff
@@ -259,30 +260,42 @@ func createNetOPNetworkInterface(
...
+		Name:      NetOPCRName(vmCtx.VM.Name, networkName, interfaceSpec.Name, true),
+	}
 
-	netIf := &netopv1alpha1.NetworkInterface{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      NetOPCRName(vmCtx.VM.Name, networkName, interfaceSpec.Name, false),
-			Namespace: vmCtx.VM.Namespace,
-		},
+	// check if a networkIf object exists with the older (v1a1) naming convention
+	if err = client.Get(vmCtx, netIfKey, netIf); ctrlruntime.IgnoreNotFound(err) != nil {
```
Comment (bryanv) [inline pkg/vmprovider/providers/vsphere2/network/network.go:273]: Move this on to its own line `err := client.Get(...)`, and remove the `err` declaration on L263 - I prefer that any vars in the if itself are scoped to just that.

Or with my later comment, maybe something like 
```
if err := client.Get(...); err != nil {
    if !apierrors.IsNotFound(err) {
        return err
    }

    name =  NetOPCRName(..., false)
}
```

File: pkg/vmprovider/providers/vsphere2/network/network.go [hunk 2]
```diff
@@ -259,30 +260,42 @@ func createNetOPNetworkInterface(
...
+		return nil, err
 	}
 
-	_, err := controllerutil.CreateOrUpdate(vmCtx, client, netIf, func() error {
```
Comment (bryanv) [inline pkg/vmprovider/providers/vsphere2/network/network.go:272]: Still always call the `CreateOrUpdate`. We want to make sure that this VM is the controlling owner of the NIC CR before using it.

File: webhooks/virtualmachine/v1alpha2/validation/virtualmachine_validator_unit_test.go [hunk 3]
```diff
@@ -260,6 +262,29 @@ func unitTestsValidateCreate() {
 			ctx.vm.Annotations[vmopv1.FirstBootDoneAnnotation] = updateSuffix
 		}
 
+		if args.extraLongNetworkIfName {
```
Comment (bryanv) [inline webhooks/virtualmachine/v1alpha2/validation/virtualmachine_validator_unit_test.go:265]: I have a strong preference that we start to move away from this struct/table driven test because it is just getting too unwieldy and hard to follow. Later in this file there's a `Context("Network") { ...` where I added newer style tests for the Network stuff - let's move these test in there.

Comment (sreyasn) [reply webhooks/virtualmachine/v1alpha2/validation/virtualmachine_validator_unit_test.go]: gotcha. Done

**Review by bryanv — APPROVED** [review]
